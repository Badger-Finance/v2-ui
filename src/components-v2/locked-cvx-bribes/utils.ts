import BigNumber from 'bignumber.js';
import { formatWithoutExtraZeros, numberWithCommas, unscale } from '../../mobx/utils/helpers';

export function formatBalance(balance: BigNumber): string {
  return numberWithCommas(formatWithoutExtraZeros(unscale(balance, 18)));
}
