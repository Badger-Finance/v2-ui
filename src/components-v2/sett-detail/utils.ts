import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/setts/sett-charts';
import { DelaySeverity } from '../../mobx/model/setts/sett-rewards';
import { SettDetailMode } from '../../mobx/model/setts/sett-detail';

dayjs.extend(utc);

export const ChartModeTitles = {
	[ChartMode.value]: 'Sett Value',
	[ChartMode.ratio]: 'Token Ratio',
	[ChartMode.accountBalance]: 'Your Holdings',
};

export const SettModeTitles = {
	[SettDetailMode.settInformation]: 'Vault Info',
	[SettDetailMode.userInformation]: 'User Info',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};
