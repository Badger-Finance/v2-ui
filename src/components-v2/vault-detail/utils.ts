import { Vault, VaultData } from '@badger-dao/sdk';
import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { ChartMode } from '../../mobx/model/vaults/vault-charts';
import { DelaySeverity } from '../../mobx/model/vaults/vault-rewards';

dayjs.extend(utc);

export const ChartModeTitles: Record<string, string> = {
	[ChartMode.Value]: 'Sett Value',
	[ChartMode.Ratio]: 'Token Ratio',
	[ChartMode.AccountBalance]: 'My Holdings',
	[ChartMode.BoostMultiplier]: 'Badger Boost',
};

export const calculateDelaySeverity = (delay: number): DelaySeverity => {
	if (delay >= 4) {
		return DelaySeverity.high;
	}

	if (delay >= 2) {
		return DelaySeverity.medium;
	}

	return DelaySeverity.none;
};

export const calculateDifferenceInHoursFromCycle = (cycle: Date): number => {
	return Math.abs(dayjs(cycle).diff(dayjs(), 'hours'));
};

export function defaultVaultBalance(vault: Vault): VaultData {
	return {
		address: vault.vaultToken,
		name: vault.name,
		symbol: vault.asset,
		pricePerFullShare: vault.pricePerFullShare,
		balance: 0,
		value: 0,
		earnedBalance: 0,
		earnedValue: 0,
		depositedBalance: 0,
		withdrawnBalance: 0,
		tokens: [],
		earnedTokens: [],
	};
}

export interface Balance {
	balance: number;
}

export function hasBalance(input?: Balance): boolean {
	if (!input) {
		return false;
	}
	return input.balance > 0;
}
