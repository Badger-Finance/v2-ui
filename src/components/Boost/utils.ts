import { makeStyles } from '@material-ui/core/styles';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import React from 'react';

type ComparisonConfig = {
  toCompareValue: number;
  toBeComparedValue: number;
  greaterCaseColor: string;
  lessCaseColor: string;
  defaultColor: string;
};

export const getColorFromComparison = ({
  toCompareValue,
  toBeComparedValue,
  greaterCaseColor,
  lessCaseColor,
  defaultColor,
}: ComparisonConfig): string => {
  if (toCompareValue > toBeComparedValue) return greaterCaseColor;
  if (toCompareValue < toBeComparedValue) return lessCaseColor;

  return defaultColor;
};

export const useAssetInputStyles = (
  currentValue: string,
  toCompare = 0,
): ((props?: React.CSSProperties) => ClassNameMap<'assetColor'>) => {
  return makeStyles((theme) => {
    const defaultColor = currentValue ? theme.palette.text.primary : theme.palette.text.secondary;
    const fontColor = getColorFromComparison({
      toCompareValue: Number(currentValue),
      toBeComparedValue: toCompare,
      greaterCaseColor: '#74D189',
      lessCaseColor: theme.palette.error.main,
      defaultColor,
    });

    return {
      assetColor: {
        color: fontColor,
      },
    };
  });
};
