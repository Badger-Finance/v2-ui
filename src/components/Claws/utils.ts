import BigNumber from 'bignumber.js';
import { INVALID_REASON } from './index';

export interface ValidationParams {
	amount: string | BigNumber;
	maximum: BigNumber;
	minimum?: BigNumber;
}

export function validateAmountBoundaries({
	amount,
	maximum,
	minimum = new BigNumber('0'),
}: ValidationParams): INVALID_REASON | undefined {
	let error: INVALID_REASON | undefined = undefined;

	const input = new BigNumber(amount);
	const amountExceedsBalance = input.gt(maximum);
	const amountIsLessThanMinimum = input.lt(minimum);

	if (amountIsLessThanMinimum) error = INVALID_REASON.UNDER_MINIMUM;
	if (amountExceedsBalance) error = INVALID_REASON.OVER_MAXIMUM;

	console.log({
		input: input.toString(),
		minimum: minimum.toString(),
		maximum: maximum.toString(),
		amountExceedsBalance,
		amountIsLessThanMinimum,
		error,
	});

	return error;
}
