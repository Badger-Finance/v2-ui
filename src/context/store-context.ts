import { createContext } from "react";
import { RootStore } from "../mobx/store";

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;