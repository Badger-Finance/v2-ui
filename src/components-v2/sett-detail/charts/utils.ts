import dayjs from 'dayjs';
import utc from 'dayjs/plugin/utc';
import { Sett } from '../../../mobx/model/setts/sett';
import { fetchSettChartInformation } from '../../../mobx/utils/apiV2';
import { SettSnapshotGranularity } from '../../../mobx/model/setts/sett-snapshot';
import { SettChartData } from '../../../mobx/model/setts/sett-charts';

dayjs.extend(utc);

export enum SettChartTimeframe {
	'day' = 1,
	'week' = 7,
	'month' = 30,
}

/**
 * Fetches chart information since the provided timeframe until the current date
 * @param sett
 * @param timeframe
 */
export const fetchSettChart = async (sett: Sett, timeframe: SettChartTimeframe): Promise<SettChartData[] | null> => {
	const timeframeDays = 1 * timeframe;
	const isDayTimeFrame = timeframe === SettChartTimeframe.day;

	// if timeframe is just one day then we want the granularity to be hours
	const granularity = isDayTimeFrame ? SettSnapshotGranularity.HOUR : SettSnapshotGranularity.DAY;

	const now = dayjs().utc(); // query until current date
	const from = dayjs(now).subtract(timeframeDays, 'days').utc();

	const fetchedData = await fetchSettChartInformation(sett.vaultToken, from.toDate(), now.toDate(), granularity);

	if (!fetchedData) {
		return null;
	}

	// data needs to be ascending sorted
	return fetchedData
		.sort((a, b) => a.timestamp - b.timestamp)
		.map((d) => ({ ...d, timestamp: new Date(d.timestamp) }));
};
