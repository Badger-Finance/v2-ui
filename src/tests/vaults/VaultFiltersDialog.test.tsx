import '@testing-library/jest-dom/extend-expect';

import { Network, Protocol, VaultBehavior, VaultState } from '@badger-dao/sdk';
import { within } from '@testing-library/react';
import UserEvent from '@testing-library/user-event';
import { StoreProvider } from 'mobx/stores/store-context';
import React from 'react';
import { config } from 'react-transition-group';

import VaultFiltersDialogV2 from '../../components-v2/VaultSearchControls/VaultFiltersDialog';
import store from '../../mobx/stores/RootStore';
import VaultStore from '../../mobx/stores/VaultStore';
import { customRender, fireEvent, screen } from '../Utils';
import { SAMPLE_VAULTS } from '../utils/samples';

describe('VaultFiltersDialog', () => {
  beforeEach(() => {
    const vaults = [...SAMPLE_VAULTS];
    vaults[0].boost.enabled = true;
    vaults[0].protocol = Protocol.Convex;
    vaults[1].protocol = Protocol.Curve;
    store.vaults.vaultCache = {
      [Network.Ethereum]: Object.fromEntries(vaults.map((vault) => [vault.vaultToken, vault])),
    };
    store.vaults.showVaultFilters = true;
    config.disabled = true;
  });

  it('renders correctly', () => {
    const { baseElement } = customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('can click checkbox filters', () => {
    customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );

    const onlyDepositsLabel = screen.getByRole('checkbox', {
      name: 'Only show deposits',
    });
    const hideDustLabel = screen.getByRole('checkbox', { name: 'Hide dust' });
    const boostedVaultsLabel = screen.getByRole('checkbox', {
      name: 'ðŸš€ Boosted Vaults',
    });

    fireEvent.click(onlyDepositsLabel);
    fireEvent.click(hideDustLabel);
    fireEvent.click(boostedVaultsLabel);

    expect(onlyDepositsLabel).toBeChecked();
    expect(hideDustLabel).toBeChecked();
    expect(boostedVaultsLabel).toBeChecked();
  });

  it('can select platforms', () => {
    customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );

    fireEvent.mouseDown(screen.getByLabelText('Platform'));
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: Protocol.Convex,
      }),
    );
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: Protocol.Curve,
      }),
    );
    expect(screen.getByTestId('platform-selector-input')).toHaveValue(Protocol.Convex + ',' + Protocol.Curve);
  });

  it('can select status', () => {
    customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );

    fireEvent.mouseDown(screen.getByLabelText('Status'));
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultState.Guarded,
      }),
    );
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultState.Experimental,
      }),
    );
    expect(screen.getByTestId('status-selector-input')).toHaveValue(VaultState.Guarded + ',' + VaultState.Experimental);
  });

  it('can select rewards', () => {
    customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );

    fireEvent.mouseDown(screen.getByLabelText('Rewards'));
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultBehavior.DCA,
      }),
    );
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultBehavior.Compounder,
      }),
    );
    expect(screen.getByTestId('rewards-selector-input')).toHaveValue(
      VaultBehavior.DCA + ',' + VaultBehavior.Compounder,
    );
  });

  it('can type search', () => {
    customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );

    fireEvent.change(screen.getByRole('textbox', { name: 'Vault Search' }), {
      target: { value: 'bitcoin convex' },
    });

    expect(screen.getByDisplayValue('bitcoin convex')).toBeInTheDocument();
  });

  it('can apply filters', async () => {
    jest.spyOn(VaultStore.prototype, 'networkHasBoostVaults', 'get').mockReturnValue(true);
    jest.spyOn(VaultStore.prototype, 'vaultsProtocols', 'get').mockReturnValue([Protocol.Convex, Protocol.Curve]);

    customRender(
      <StoreProvider value={store}>
        <VaultFiltersDialogV2 />
      </StoreProvider>,
    );

    fireEvent.click(screen.getByRole('checkbox', { name: 'Only show deposits' }));
    fireEvent.click(screen.getByRole('checkbox', { name: 'Hide dust' }));
    fireEvent.click(screen.getByRole('checkbox', { name: 'ðŸš€ Boosted Vaults' }));

    fireEvent.mouseDown(screen.getByLabelText('Platform'));
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: Protocol.Convex,
      }),
    );
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: Protocol.Curve,
      }),
    );
    UserEvent.type(screen.getByRole('listbox'), '{esc}');

    fireEvent.mouseDown(screen.getByLabelText('Status'));
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultState.Guarded,
      }),
    );
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultState.Experimental,
      }),
    );
    UserEvent.type(screen.getByRole('listbox'), '{esc}');

    fireEvent.mouseDown(screen.getByLabelText('Rewards'));
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultBehavior.DCA,
      }),
    );
    fireEvent.click(
      within(screen.getByRole('listbox')).getByRole('option', {
        name: VaultBehavior.Compounder,
      }),
    );
    UserEvent.type(screen.getByRole('listbox'), '{esc}');

    fireEvent.change(screen.getByRole('textbox', { name: 'Vault Search' }), {
      target: { value: 'bitcoin convex' },
    });
    fireEvent.click(screen.getByRole('button', { name: 'Apply Filters' }));

    expect(store.vaults.vaultsFilters).toMatchSnapshot();
  });
});
