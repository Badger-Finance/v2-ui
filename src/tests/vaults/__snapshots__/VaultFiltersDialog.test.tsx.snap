// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VaultFiltersDialog can apply filters 1`] = `
Object {
  "behaviors": Array [
    "DCA",
    "Compounder",
  ],
  "currency": "BTC",
  "hidePortfolioDust": true,
  "onlyBoostedVaults": true,
  "onlyDeposits": true,
  "protocols": Array [
    "Convex",
    "Curve",
  ],
  "search": "bitcoin convex",
  "showAPR": true,
  "statuses": Array [
    "guarded",
    "experimental",
  ],
}
`;

exports[`VaultFiltersDialog renders correctly 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
   
   
  
  
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root-8"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root-23"
      style="opacity: 1;"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container-11 MuiDialog-scrollPaper-9"
      role="none presentation"
      style="opacity: 1;"
      tabindex="-1"
    >
      <div
        class="MuiPaper-root-25 MuiDialog-paper-12 MuiDialog-paperScrollPaper-13 MuiDialog-paperWidthSm-17 MuiDialog-paperFullWidth-21 MuiPaper-elevation24-52 MuiPaper-rounded-26"
        role="dialog"
      >
        <div
          class="MuiDialogTitle-root-53 makeStyles-title-1"
        >
          <div
            class="MuiGrid-root-54 MuiGrid-container-55 MuiGrid-align-items-xs-center-63 MuiGrid-justify-content-xs-space-between-74"
          >
            <h5
              class="MuiTypography-root-157 MuiTypography-h5-166 MuiTypography-displayInline-185"
            >
              Filter & Search
            </h5>
            <button
              aria-label="close vault filters"
              class="MuiButtonBase-root-196 MuiIconButton-root-187 makeStyles-closeButton-3"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label-195"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root-199"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root-383"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiDialogContent-root-208 makeStyles-content-2"
        >
          <hr
            class="MuiDivider-root-210 makeStyles-topDivider-6"
          />
          <div
            class="MuiGrid-root-54 MuiGrid-container-55 MuiGrid-direction-xs-column-58"
          >
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <label
                class="MuiFormControlLabel-root-217"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root-196 MuiIconButton-root-187 PrivateSwitchBase-root-229 MuiCheckbox-root-223 MuiCheckbox-colorPrimary-227 MuiIconButton-colorPrimary-191"
                >
                  <span
                    class="MuiIconButton-label-195"
                  >
                    <input
                      class="PrivateSwitchBase-input-232"
                      data-indeterminate="false"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-199"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </span>
                <span
                  class="MuiTypography-root-157 MuiFormControlLabel-label-222 MuiTypography-body1-159"
                >
                  <p
                    class="MuiTypography-root-157 MuiTypography-body2-158"
                  >
                    Only show deposits
                  </p>
                </span>
              </label>
            </div>
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <label
                class="MuiFormControlLabel-root-217"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root-196 MuiIconButton-root-187 PrivateSwitchBase-root-229 MuiCheckbox-root-223 MuiCheckbox-colorPrimary-227 MuiIconButton-colorPrimary-191"
                >
                  <span
                    class="MuiIconButton-label-195"
                  >
                    <input
                      class="PrivateSwitchBase-input-232"
                      data-indeterminate="false"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-199"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </span>
                <span
                  class="MuiTypography-root-157 MuiFormControlLabel-label-222 MuiTypography-body1-159"
                >
                  <p
                    class="MuiTypography-root-157 MuiTypography-body2-158"
                  >
                    Hide dust
                  </p>
                </span>
              </label>
            </div>
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <label
                class="MuiFormControlLabel-root-217 makeStyles-label-233"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root-196 MuiIconButton-root-187 PrivateSwitchBase-root-229 MuiCheckbox-root-223 MuiCheckbox-colorPrimary-227 MuiIconButton-colorPrimary-191"
                >
                  <span
                    class="MuiIconButton-label-195"
                  >
                    <input
                      class="PrivateSwitchBase-input-232"
                      data-indeterminate="false"
                      type="checkbox"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root-199"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </span>
                <span
                  class="MuiTypography-root-157 MuiFormControlLabel-label-222 MuiTypography-body1-159"
                >
                  <p
                    class="MuiTypography-root-157 MuiTypography-body2-158"
                  >
                    ðŸš€ Boosted Vaults
                  </p>
                </span>
              </label>
            </div>
          </div>
          <hr
            class="MuiDivider-root-210 makeStyles-divider-5"
          />
          <div
            class="MuiGrid-root-54 MuiGrid-container-55 MuiGrid-direction-xs-column-58"
          >
            <div
              class="MuiGrid-root-54 makeStyles-select-4 MuiGrid-item-56"
            >
              <div
                class="MuiFormControl-root-236 makeStyles-formControl-234"
              >
                <label
                  class="MuiFormLabel-root-252 MuiInputLabel-root-240 MuiInputLabel-formControl-246 MuiInputLabel-animated-249 MuiInputLabel-outlined-251"
                  data-shrink="false"
                  id="status-selector-id-label"
                >
                  Status
                </label>
                <div
                  class="MuiInputBase-root-286 MuiOutlinedInput-root-271 MuiInputBase-formControl-287"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="status-selector-id-label status-selector"
                    class="MuiSelect-root-260 MuiSelect-select-261 MuiSelect-selectMenu-264 MuiSelect-outlined-263 MuiInputBase-input-297 MuiOutlinedInput-input-281"
                    id="status-selector"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      â€‹
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput-270"
                    data-testid="status-selector-input"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-199 MuiSelect-icon-266 MuiSelect-iconOutlined-269"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-308 MuiOutlinedInput-notchedOutline-280"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-310"
                    >
                      <span>
                        Status
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-54 makeStyles-select-4 MuiGrid-item-56"
            >
              <div
                class="MuiFormControl-root-236 makeStyles-formControl-312"
              >
                <label
                  class="MuiFormLabel-root-252 MuiInputLabel-root-240 MuiInputLabel-formControl-246 MuiInputLabel-animated-249 MuiInputLabel-outlined-251"
                  data-shrink="false"
                  id="platform-selector-id-label"
                >
                  Platform
                </label>
                <div
                  class="MuiInputBase-root-286 MuiOutlinedInput-root-271 MuiInputBase-formControl-287"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="platform-selector-id-label platform-selector"
                    class="MuiSelect-root-260 MuiSelect-select-261 MuiSelect-selectMenu-264 MuiSelect-outlined-263 MuiInputBase-input-297 MuiOutlinedInput-input-281"
                    id="platform-selector"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      â€‹
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput-270"
                    data-testid="platform-selector-input"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-199 MuiSelect-icon-266 MuiSelect-iconOutlined-269"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-308 MuiOutlinedInput-notchedOutline-280"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-310"
                    >
                      <span>
                        Platform
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-54 makeStyles-select-4 MuiGrid-item-56"
            >
              <div
                class="MuiFormControl-root-236 makeStyles-formControl-314"
              >
                <label
                  class="MuiFormLabel-root-252 MuiInputLabel-root-240 MuiInputLabel-formControl-246 MuiInputLabel-animated-249 MuiInputLabel-outlined-251"
                  data-shrink="false"
                  id="rewards-selector-id-label"
                >
                  Rewards
                </label>
                <div
                  class="MuiInputBase-root-286 MuiOutlinedInput-root-271 MuiInputBase-formControl-287"
                >
                  <div
                    aria-haspopup="listbox"
                    aria-labelledby="rewards-selector-id-label rewards-selector"
                    class="MuiSelect-root-260 MuiSelect-select-261 MuiSelect-selectMenu-264 MuiSelect-outlined-263 MuiInputBase-input-297 MuiOutlinedInput-input-281"
                    id="rewards-selector"
                    role="button"
                    tabindex="0"
                  >
                    <span>
                      â€‹
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput-270"
                    data-testid="rewards-selector-input"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root-199 MuiSelect-icon-266 MuiSelect-iconOutlined-269"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-308 MuiOutlinedInput-notchedOutline-280"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-310"
                    >
                      <span>
                        Rewards
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <div
                class="MuiFormControl-root-236 MuiTextField-root-318 makeStyles-root-316"
              >
                <div
                  class="MuiInputBase-root-286 MuiOutlinedInput-root-271 MuiInputBase-formControl-287 MuiInputBase-adornedEnd-291 MuiOutlinedInput-adornedEnd-276"
                >
                  <input
                    aria-invalid="false"
                    aria-label="Vault Search"
                    class="MuiInputBase-input-297 MuiOutlinedInput-input-281 MuiInputBase-inputAdornedEnd-302 MuiOutlinedInput-inputAdornedEnd-285"
                    placeholder="Search by vault name, token, rewards..."
                    type="text"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root-319 MuiInputAdornment-positionEnd-322"
                  >
                    <button
                      aria-label="search"
                      class="MuiButtonBase-root-196 MuiIconButton-root-187 makeStyles-icon-317 MuiIconButton-sizeSmall-194"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label-195"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root-199"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root-383"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-308 MuiOutlinedInput-notchedOutline-280"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-309"
                      style="width: 0.01px;"
                    >
                      <span>
                        â€‹
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root-210 makeStyles-divider-5"
          />
          <div
            class="MuiGrid-root-54 MuiGrid-container-55"
          >
            <div
              class="MuiGrid-root-54 makeStyles-apr-7 MuiGrid-item-56"
            >
              <div
                aria-label="apy apr selector"
                class="MuiButtonGroup-root-328"
                role="group"
              >
                <button
                  aria-selected="true"
                  class="MuiButtonBase-root-196 MuiButton-root-352 MuiButton-outlined-357 MuiButtonGroup-grouped-334 MuiButtonGroup-groupedHorizontal-335 MuiButtonGroup-groupedOutlined-342 MuiButtonGroup-groupedOutlinedHorizontal-343 MuiButtonGroup-groupedOutlined-342 makeStyles-selected-326 makeStyles-button-327"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label-353"
                  >
                    APY
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </button>
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root-196 MuiButton-root-352 MuiButton-outlined-357 MuiButtonGroup-grouped-334 MuiButtonGroup-groupedHorizontal-335 MuiButtonGroup-groupedOutlined-342 MuiButtonGroup-groupedOutlinedHorizontal-343 MuiButtonGroup-groupedOutlined-342 makeStyles-button-327"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label-353"
                  >
                    APR
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <div
                aria-label="currency selector"
                class="MuiButtonGroup-root-328"
                role="group"
              >
                <button
                  aria-selected="true"
                  class="MuiButtonBase-root-196 MuiButton-root-352 MuiButton-outlined-357 MuiButtonGroup-grouped-334 MuiButtonGroup-groupedHorizontal-335 MuiButtonGroup-groupedOutlined-342 MuiButtonGroup-groupedOutlinedHorizontal-343 MuiButtonGroup-groupedOutlined-342 makeStyles-selected-381 makeStyles-button-382"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label-353"
                  >
                    USD
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </button>
                <button
                  aria-selected="false"
                  class="MuiButtonBase-root-196 MuiButton-root-352 MuiButton-outlined-357 MuiButtonGroup-grouped-334 MuiButtonGroup-groupedHorizontal-335 MuiButtonGroup-groupedOutlined-342 MuiButtonGroup-groupedOutlinedHorizontal-343 MuiButtonGroup-groupedOutlined-342 makeStyles-button-382"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiButton-label-353"
                  >
                    BTC
                  </span>
                  <span
                    class="MuiTouchRipple-root-383"
                  />
                </button>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root-210 makeStyles-divider-5"
          />
          <div
            class="MuiGrid-root-54 MuiGrid-container-55 MuiGrid-spacing-xs-2-78 MuiGrid-justify-content-xs-flex-end-73"
          >
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <button
                class="MuiButtonBase-root-196 MuiButton-root-352 MuiButton-text-354 MuiButton-textPrimary-355"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-353"
                >
                  Clear All
                </span>
                <span
                  class="MuiTouchRipple-root-383"
                />
              </button>
            </div>
            <div
              class="MuiGrid-root-54 MuiGrid-item-56"
            >
              <button
                class="MuiButtonBase-root-196 MuiButton-root-352 MuiButton-contained-360 MuiButton-containedPrimary-361"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label-353"
                >
                  Apply Filters
                </span>
                <span
                  class="MuiTouchRipple-root-383"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
